docs/tutorials/azure.md

```diff
@@ -0,0 +1,166 @@
+# Setting up ExternalDNS for Services on Azure
+
+This tutorial describes how to setup ExternalDNS for usage within a Kubernetes cluster on Azure.
+
+Make sure to use **>=0.4** version of ExternalDNS for this tutorial.
+
+This tutorial uses [Azure CLI 2.0](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) for all
+Azure commands and assumes that the Kubernetes cluster was created via Azure Container Services and `kubectl` commands
+are being run on an orchestration master.
+
+## Creating a Azure DNS zone
+
+The Azure provider for ExternalDNS will find suitable zones for domains it manages; it will
+not automatically create zones.
+
+For this tutorial, we will create a Azure resource group named 'externaldns' that can easily be deleted later:
+
+```
+$ az group create -n externaldns -l eastus
+```
+
+Substitute a more suitable location for the resource group if desired.
+
+Next, create a Azure DNS zone for "example.com":
+
+```
+$ az network dns zone create -g externaldns -n example.com
+```
+
+Substitute a domain you own for "example.com" if desired.
+
+If using your own domain that was registered with a third-party domain registrar, you should point your domain's
+name servers to the values in the `nameServers` field from the JSON data returned by the `az network dns zone create` command.
+Please consult your registrar's documentation on how to do that.
+
+## Creating Azure Credentials Secret
+
+When your Kubernetes cluster is created by Azure Container Services, a file named `/etc/kubernetes/azure.json` is created to store
+the Azure credentials for API access.  Kubernetes uses this file for the Azure cloud provider.
+
+For ExternalDNS to access the Azure API, it also needs access to this file.  However, we will be deploying ExternalDNS inside of
+the Kubernetes cluster so we will need to use a Kubernetes secret.
+
+To create the secret:
+
+```
+$ kubectl create secret generic azure-credentials-secret --from-file=/etc/kubernetes/azure.json
+```
+
+## Deploy ExternalDNS
+
+Create a deployment file called `externaldns.yaml` with the following contents:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:v0.4.0
+        args:
+        - --source=service
+        - --domain-filter=example.com. # (optional) limit to only example.com domains; change to match the zone created above.
```

Comment:
@ideahitme are ou saying this depends on how the DNS zone was created? This would be strange.